smart-common/src/main/java/org/smartdata/common/models/FileInfo.java

```diff
@@ -165,6 +165,27 @@ public static FileInfo fromHdfsFileStatus(HdfsFileStatus status, String parent)
         .build();
   }
 
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+
+    FileInfo fileInfo = (FileInfo) o;
+
+    if (fileId != fileInfo.fileId) return false;
```

Comment:
combine some of the comparisons together,  and using {} even it contains only 1 statement